### PR TITLE
499-refactor: Widget alumni

### LINKED ISSUE #499: FSD: Widget Alumni
## 📝 Describe the refactoring task

Part of the issue #347 
Alumni component refactor

## 📍 Code location

`src/widgets/alumni`

## ✅ Definition of Done (DoD):

1. Refactor `scss` to `scss modules` as following:

-   remove all parent selector `& .card-content` -> `.card-content`. 
-   change all possible values to variables if such exists.
-   rename all classes to be generic e.g `.nodejs-trainer 🙅‍♂️` -> `.trainer 👌`.
-   and everything in your opinion that needs to be refactored

3. delete `AlumniProps type`
4. in the `constants.ts` remove type annotation from `alumni` and change it as following
``` typescript
const alumni = [...] as const
```
5. move unit test inside the `ui` folder
6. move all the `magic numbers` in the constants and place these constants inside the `constants.ts` folder

### PR DESCRIPTION
**Pull request by Wystov**
## What type of PR is this? (select all that apply)

- [x] 🧑‍💻 Code Refactor

## Description
- converted css to scss modules + classNames lib
- changed flex to grid (it looked and behave like grid, so i decided that using grid here would be more straightforward and readable)
- changed in component array length manipulations for different screen width to media queries in scss file.
- removed unit tests related to deleted functionality and updated playwright tests

## Related Tickets & Documents

- Related Issue #499 
- Closes #499 

## Screenshots, Recordings

_Please replace this line with any relevant images for UI changes._

## Added/updated tests?

- [x] 👌 Yes

## [optional] What gif best describes this PR or how it makes you feel?

doc_2024-09-06_20-19-47

## Summary by CodeRabbit

- **New Features**
	- Introduced a responsive design for the alumni interface with new styling classes.
	- Added unit tests for the Alumni component to ensure correct rendering of key elements.

- **Bug Fixes**
	- Removed unnecessary complexity from the Alumni component, ensuring it now displays the entire alumni list consistently.

- **Refactor**
	- Updated styling approach to use CSS modules for better modularity and maintainability.
	- Enhanced type safety by simplifying type declarations for the alumni constant.

### CODE REVIEW

**Review by Quiddlee — APPROVED** [review]
🔥🔥🔥

File: src/widgets/alumni/ui/alumni.module.scss [hunk 1]
```diff
@@ -1,14 +1,16 @@
-h2,
-p {
-  width: 640px;
-
-  @include media-tablet {
-    width: 100%;
+.info-wrapper {
+  h2,
```
[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/alumni/ui/alumni.module.scss:2]: It was like that in old styles. Btw how can we have a long title here if it's one purpose component and title is already hardcoded in it?

[the remark this answers is not in the document]
Comment (ansivgit) [reply src/widgets/alumni/ui/alumni.module.scss:2]: That's why I suggest removing the fixed width for `h2` - it's redundant

[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/alumni/ui/alumni.module.scss:2]: done

File: src/widgets/alumni/ui/alumni.module.scss [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+    @include media-tablet {
+      width: 100%;
+    }
   }
-}
 
-p {
-  margin-top: 16px;
+  p {
+    margin-top: 16px;
```
[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/alumni/ui/alumni.module.scss:12]: as i see in other sections there is both padding bottom in title and margin - top from paragraph (or same value gap). i f i'll remove mt here spacings will be inconsistent between sections

File: src/widgets/alumni/ui/alumni.module.scss [hunk 3]
```diff
@@ -18,6 +20,18 @@ p {
   justify-content: space-between;
 
   margin-top: 40px;
```
[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/alumni/ui/alumni.module.scss:22]: changed

File: src/widgets/alumni/ui/alumni.module.scss [hunk 4]
```diff
@@ -18,6 +20,18 @@ p {
...
+  }
+
+  @include media-tablet {
+    .logo-container:nth-child(n + 7) {
+      display: none;
+    }
+  }
 }
 
 .logo-container {
```
[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/alumni/ui/alumni.module.scss]: changed

File: src/widgets/alumni/ui/alumni.test.tsx [hunk 5]
```diff
@@ -1,15 +1,6 @@
...
-
-vi.mock('@/shared/hooks/use-window-size', () => ({
-  useWindowSize: vi.fn().mockReturnValue({
-    width: 2000,
-    height: 2000,
-  }),
-}),
-);
 
 describe('Alumni', () => {
```
[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/alumni/ui/alumni.test.tsx:5]: added

**Comment by SpaNb4** [comment]
run visual now

**Review by SpaNb4 — APPROVED** [review]

**Comment by SpaNb4** [comment]
run visual now

File: src/widgets/alumni/ui/alumni.tsx [hunk 6]
```diff
@@ -1,41 +1,26 @@
...
-    alumniArr = alumni.slice(0, 6);
-  } else {
-    alumniArr = [...alumni];
-  }
-
   return (
-    <article className="alumni container">
-      <section className="alumni content">
+    <article className={cx('container')}>
+      <section className={cx('content', 'info-wrapper')}>
```
Comment (ansivgit) [inline src/widgets/alumni/ui/alumni.tsx:14]:
```suggestion
      <section className={cx('content', 'alumni')}>
```

File: src/widgets/alumni/ui/alumni.tsx [hunk 7]
```diff
@@ -1,41 +1,26 @@
...
-        <Paragraph className="alumni-paragraph">
+        <Paragraph>
           We are immensely proud of RS School alumni who build their successful careers in ambitious
           IT companies
         </Paragraph>
-        <section className="alumni">
-          {alumniArr.map(({ id, image }) => (
-            <figure key={id} className="alumni-logo-container">
-              <Image className="alumni-logo" src={image} alt={id} />
+        <section className={cx('alumni')}>
```
Comment (ansivgit) [inline src/widgets/alumni/ui/alumni.tsx:20]:
```suggestion
        <section className={cx('alumni-list')}>
```

File: src/widgets/alumni/ui/alumni.tsx (hunk 6 above)
Comment (ansivgit) [reply src/widgets/alumni/ui/alumni.tsx:14]: I suggest using more specific classes for sections, class `info-wrapper` is used in several widgets, using more specific names will make it easier to navigate the code base

**Review by ansivgit — APPROVED** [review]

**Pull request merged**